rules/prefer-object-define-properties.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+const isSameReference = require('./utils/is-same-reference.js');
+const {removeArgument} = require('./fix/index.js');
+
+const MESSAGE_ID = 'prefer-object-define-properties';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` over multiple `{{value}}`.',
+};
+
+function getFirstExpression(node, sourceCode) {
+	const {parent} = node;
+	const visitorKeys = sourceCode.visitorKeys[parent.type] || Object.keys(parent);
+
+	for (const property of visitorKeys) {
+		const value = parent[property];
+		if (Array.isArray(value)) {
+			const index = value.indexOf(node);
+
+			if (index !== -1) {
+				return value[index - 1];
+			}
+		}
+	}
+}
+
+function isObjectDefineProperty(node) {
+	return (
+		node
+		&& node.type === 'CallExpression'
+		&& node.callee.type === 'MemberExpression'
+		&& node.callee.object.name === 'Object'
+		&& /definePropert(y|ies)/.test(node.callee.property.name)
+	);
+}
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => ({
+	'Program:exit'() {
+		const sourceCode = context.getSourceCode();
+		const references = context
+			.getScope()
+			.variableScope.set.get('Object')
+			.references.filter(
+				reference =>
+					(reference.identifier.parent.type === 'MemberExpression'
+						&& reference.identifier.parent.property.name === 'defineProperty'
+						&& /Identifier|Literal/.test(reference.identifier.parent.parent.arguments?.[1]?.type)
+						&& /Identifier|ObjectExpression/.test(reference.identifier.parent.parent.arguments?.[2]?.type))
+					|| (reference.identifier.parent.property.name === 'defineProperties' && /Identifier|ObjectExpression/.test(reference.identifier.parent.parent.arguments?.[1]?.type)),
+			)
+			.map(reference => ({
+				scope: reference.from.block,
+				node: reference.identifier.parent.parent,
+				arguments: reference.identifier.parent.parent.arguments,
+			}));
+
+		const groups = [];
+		let currentGroup;
+
+		for (const reference of references) {
+			const firstExpression = getFirstExpression(reference.node.parent, sourceCode);
```

Comment:
I didn't see `ExpressionStatement` check https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1843#discussion_r909226611, let's only check `ExpressionStatement`.

Reply:
Need test, you can use https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1843#discussion_r909226611